lib/routes/create_invoice/pay_invoice_page.dart

```diff
@@ -0,0 +1,393 @@
+          userInput: true),
+    );
+  }
+
+  void applyPayFetchResponse(PayFetchResponse response, AccountModel account) {
+    _payFetchResponse = response;
+    _commentController.text = response.comment;
+    _amountController.text = account.currency
+        .format(response.minAmount, includeDisplayName: false, userInput: true);
+  }
+
+  String _getMetadataText(
+      /* List<List<String>>*/ List<LNURLPayResponse1_Metadata> metadata) {
+    /*
+
+    `metadata` json array must contain one `text/plain` entry, all other types of entries are optional:
+
+    ```
+    [
+        [
+            "text/plain", // must always be present
+            content // actual metadata content
+        ],
+        [
+            "image/png;base64", // optional 512x512px PNG thumbnail which will represent this lnurl in a list or grid
+            content // base64 string, up to 136536 characters (100Kb of image data in base-64 encoding)
+        ],
+        [
+            "image/jpeg;base64", // optional 512x512px JPG thumbnail which will represent this lnurl in a list or grid
+            content // base64 string, up to 136536 characters (100Kb of image data in base-64 encoding)
+        ],
+        ... // more objects for future types
+    ]
+    ```
+
+    and be sent as a string:
+
+    ```
+    "[[\"text/plain\", \"lorem ipsum blah blah\"]]"
+    ```
+    */
+    var result = '';
+    log.info('_getMetadataText: metadata: $metadata');
+
+    for (var datum in metadata) {
+      if (datum.entry[0] == 'text/plain') {
+        result = datum.entry[1];
+        break;
+      }
+    }
+
+    return result;
+  }
+
+  void _getInvoice(InvoiceBloc invoiceBloc, AccountBloc accountBloc,
+      LNUrlBloc lnurlBloc, AccountModel account) async {
+    log.info('_getInvoice.');
+    /*
+         5. LN WALLET makes a GET request using callback with the following query parameters:
+         amount (input) - user specified sum in MilliSatoshi
+         nonce - an optional parameter used to prevent server response caching
+         fromnodes - an optional parameter with value set to comma separated nodeIds if payer wishes a service to provide payment routes starting from specified LN nodeIds
+         comment (input) - an optional parameter to pass the LN WALLET user's comment to LN SERVICE
+
+    */
+
+    _payFetchResponse.amount =
+        account.currency.parse(_amountController.text) * 1000;
+    _payFetchResponse.comment = _commentController.text;
+
+    LNUrlPayInfo payinfo;
+    var action = Pay(_payFetchResponse);
+    lnurlBloc.actionsSink.add(action);
+    action.future.catchError((e) {
+      promptError(
+        context,
+        "An error occurred while attempting to retreive an invoice from ${_payFetchResponse.host}!",
+        Text("Reason: ${e.toString()}",
+            style: Theme.of(context).dialogTheme.contentTextStyle),
+      );
+    }).then((payinfo) {
+      var navigator = Navigator.of(context);
+      navigator.pop();
```

Comment:
Before popping here, I suggest to check if the user cancelled by closing the dialog (back on android).
One way to do it is to check if the current widget is disposed and continue with the invoice only if it is not.